test/controllers/CyaMaintainProfileControllerSpec.scala

```diff
@@ -0,0 +1,287 @@
+              verify(mockTraderProfileConnector)
+                .submitTraderProfile(eqTo(updatedTraderProfile), eqTo(testEori))(any())
+            }
+
+            withClue("must call the audit connector with the supplied details") {
+              verify(mockAuditService)
+                .auditMaintainProfile(eqTo(traderProfile), eqTo(updatedTraderProfile), eqTo(AffinityGroup.Individual))(
+                  any()
+                )
+            }
+          }
+        }
+
+        "when user answers cannot remove Nirms and cannot update user profile" - {
+
+          "if the data is invalid then must not submit anything, and redirect to Journey Recovery" - {
+
+            val userAnswers = emptyUserAnswers
+              .set(RemoveNirmsPage, true)
+              .success
+              .value
+              .set(HasNirmsUpdatePage, true)
+              .success
+              .value
+
+            val mockTraderProfileConnector = mock[TraderProfileConnector]
+            val mockAuditService           = mock[AuditService]
+
+            val application = applicationBuilder(userAnswers = Some(userAnswers))
+              .overrides(
+                bind[TraderProfileConnector].toInstance(mockTraderProfileConnector),
+                bind[AuditService].toInstance(mockAuditService)
+              )
+              .build()
+
+            running(application) {
+
+              val request = FakeRequest(POST, routes.CyaMaintainProfileController.onSubmitNirms.url)
+
+              val result = route(application, request).value
+
+              status(result) mustEqual SEE_OTHER
+              redirectLocation(result).value mustEqual
+                routes.JourneyRecoveryController.onPageLoad(Some(RedirectUrl(journeyRecoveryContinueUrl))).url
+
+              withClue("must not call the trader profile connector") {
+                verify(mockTraderProfileConnector, never()).getTraderProfile(any())(any())
+              }
+
+              withClue("must not call the audit connector") {
+                verify(mockAuditService, never()).auditMaintainProfile(any(), any(), any())(any())
+              }
+            }
+
+          }
+
+          "must not submit anything, and redirect to Journey Recovery" in {
+
+            val application =
+              applicationBuilder(userAnswers = Some(emptyUserAnswers))
+                .build()
+
+            running(application) {
+              val request = FakeRequest(POST, routes.CyaMaintainProfileController.onSubmitNirms.url)
+
+              val result = route(application, request).value
+
+              status(result) mustEqual SEE_OTHER
+              redirectLocation(result).value mustEqual
+                routes.JourneyRecoveryController.onPageLoad(Some(RedirectUrl(journeyRecoveryContinueUrl))).url
+            }
+
+          }
+        }
+
+        "must let the play error handler deal with connector failure" in {
```

Comment:
Test case: if the Submit fails you still should have audited.

Reply:
New test added https://github.com/hmrc/trader-goods-profiles-frontend/pull/404/files#diff-aa09ec45c5a5564873c1cdc9c5b6c0c2a57b149f130d3fb3568f1f348cbcbf8cR288